src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs

```diff
@@ -68,23 +68,45 @@ public static void ReadAfterInitializationWithTextReaderOnAsyncReaderDoesNotThro
 
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap |  //[ActiveIssue(13121)]  // Path cannot be resolved in UWP
-        TargetFrameworkMonikers.NetFramework)]    // Full framework uses XmlUrlResolver instead of SystemPathResolver (which doesn't allow access to this path)
+        TargetFrameworkMonikers.NetFramework)]  //[ActiveIssue(19806)] // Full framework uses WebRequest instead of HttpRequest
         public static void ReadAsyncAfterInitializationWithUriThrows()
         {
             using (XmlReader reader = XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = true }))
             {
-                Assert.Throws<XmlException>(() => reader.ReadAsync().GetAwaiter().GetResult());
+                Assert.Throws<System.Net.Http.HttpRequestException>(() => reader.ReadAsync().GetAwaiter().GetResult());
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)] // Full framework uses WebRequest instead of HttpRequest
+        public static void ReadAsyncAfterInitializationWithUriThrows_FullFramework()
+        {
+            using (XmlReader reader = XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = true }))
+            {
+                Assert.Throws<System.Net.WebException>(() => reader.ReadAsync().GetAwaiter().GetResult());
```

Comment:
@stephentoub Yes. I'm going to remove this difference and make Core match Desktop in another PR when this PR gets merged, so we can leave these as is. But out of curiousity, how can the extensions help here? Can you please give an example?